components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/filter/AuthorizationHeaderFilter.java

```diff
@@ -0,0 +1,131 @@
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+
+/**
+ * Servlet Filter implementation class for authorization header validation. This handles the NPE thrown by CXF
+ * transport exposing the stack-trace and server information to the end user for authorization headers with invalid
+ * encoded json data by sending a proper error response back to the user. If there is more than single space between
+ * auth_type and json data this will modify the header to be in correct format and pass the request along the filter
+ * chain.
+ *
+ * @since 4.7.6
+ */
+@WebFilter("/AuthorizationHeaderFilter")
+public class AuthorizationHeaderFilter implements Filter {
+
+    private static final String AUTH_TYPE_BASIC = "Basic";
+    private static final int AUTH_TYPE_BASIC_LENGTH = 5; //length of "Basic" String
+    private static final int MINIMUM_CREDENTIAL_SIZE = 4;
+
+    private ServletContext context;
+
+    public void init(FilterConfig fConfig) throws ServletException {
+
+        this.context = fConfig.getServletContext();
+        this.context.log("AuthorizationHeaderFilter initialized");
+    }
+
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        final HttpServletRequest req = (HttpServletRequest) request;
+        final String authHeader = req.getHeader(HTTPConstants.HEADER_AUTHORIZATION);
+        if (StringUtils.isEmpty(authHeader)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        String authType = null;
+        if (authHeader.length() >= AUTH_TYPE_BASIC_LENGTH) {
+            authType = authHeader.trim().substring(0, AUTH_TYPE_BASIC_LENGTH);
+        }
+        if (AUTH_TYPE_BASIC.equals(authType)) {
+            final String authCredentials = (authHeader.trim().substring(AUTH_TYPE_BASIC_LENGTH)).trim();
+            //if auth header comes in invalid format send error in response
+            if (StringUtils.isBlank(authCredentials) || authCredentials.indexOf(' ') >= 0
+                    || authCredentials.length() < MINIMUM_CREDENTIAL_SIZE) {
+                String errorMsg = "Internal Server Error";
+                handleErrorResponse((HttpServletResponse) response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                        errorMsg);
+            } else if (authHeader.substring((AUTH_TYPE_BASIC + " ").length()).startsWith(" ")) {
+                //if there is more than single space between auth_type and credentials modify the request header
+                HttpServletRequestWrapper wrapper = new HttpServletRequestWrapper(req) {
+
```

Comment:
unnecessary new line

Reply:
newline added when content is properly formatted